### PR TITLE
Fix installation on mac

### LINKED ISSUE #356: make install fails on MacOS
`sudo make install` fails on MacOS 14.6, because installing brew packages as a root is no longer supported

Screenshot 2024-09-03 at 11 36 49

### PR DESCRIPTION
**Pull request by nai1ka**
* Fixes error with brew packages installation, when using sudo (see #356)
* Fixes error with pip packages (see #356)

### CODE REVIEW

File: steps/install.sh [hunk 1]
```diff
@@ -59,15 +59,15 @@ install_package xmlstarlet
...
+  sudo -u $SUDO_USER  brew install coreutils
+  sudo -u $SUDO_USER  brew install wget
 fi
 
 if ! pdftotext -v >/dev/null 2>&1; then
   if "${LOCAL}/help/is-linux.sh"; then
     "${LOCAL}/help/sudo.sh" apt-get install -y xpdf
   elif "${LOCAL}/help/is-macos.sh"; then
-    brew install poppler
+    sudo -u $SUDO_USER  brew install poppler
```
Comment (yegor256) [inline steps/install.sh:70]: @nai1ka maybe it's better to use `"${LOCAL}/help/sudo.sh"`, as we do in all other places?

File: installs/install-pip.sh [hunk 2]
```diff
@@ -27,5 +27,13 @@ set -o pipefail
 "${LOCAL}/help/assert-tool.sh" python3 --version
 "${LOCAL}/help/assert-tool.sh" pip3 --version
 
+# Activate the virtual environment
+if [ ! -d "${LOCAL}/venv" ]; then
+    python3 -m venv "${LOCAL}/venv"
+fi
+
+# Activate the virtual environment
+source "${LOCAL}/venv/bin/activate"
```
Comment (yegor256) [inline installs/install-pip.sh:36]: @nai1ka please, merge `master` branch into your branch, some changes have already been made

File: steps/install.sh (hunk 1 above)
Comment (nai1ka) [reply steps/install.sh]: Sure, fixed this.
Now sudo.sh can be used with --as-user flag in order to run commands as a default user

**Comment by nai1ka** [comment]
@yegor256 Done, "resolved" all merge conflicts :)

**Comment by yegor256** [comment]
@nai1ka thanks for this PR! However, we still have broken CI pipelines. Maybe you know how to fix them?

**Pull request merged**